card-games-vite/src/utils/deckOfCardsApi.js

```diff
@@ -0,0 +1,84 @@
+import { checkResponse } from "./api";
+
+const deckOfCardsUrl = "https://deckofcardsapi.com";
+
+function createNewDeck(deckNumber) {
+  return fetch(
+    `https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=${deckNumber}`
+  ).then(checkResponse);
```

Comment:
**COULD BE IMPROVED**

You can make a special function for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.